src/scitacean/transfer/ssh.py

```diff
@@ -95,18 +96,18 @@ def _sftp(self) -> SFTPClient:
         return self._connection.sftp()
 
     @property
-    def source_dir(self) -> str:
-        return os.path.join(self._remote_base_path, self._dataset_id.pid)
+    def source_dir(self) -> RemotePath:
+        return self._remote_base_path / self._dataset_id.pid
 
-    def remote_path(self, filename) -> str:
-        return os.path.join(self.source_dir, filename)
+    def remote_path(self, filename) -> RemotePath:
+        return self.source_dir / filename
 
     def _make_source_folder(self):
         try:
-            self._sftp.mkdir(self.source_dir)
+            self._connection.run(f"mkdir -p {os.fspath(self.source_dir)}", hide=True)
```

Comment:
Would this work only on Linux/Mac?

Reply:
For the remote? Yes. But see my comment in the PR. The whole path handling assumes a UNIX (like) server.

I made the change here because `sftp.mkdir` does not create parents.

Reply:
Ok!